### PR TITLE
Component: Input Type Select PR

### PR DESCRIPTION
**Pull request by torykling**
#20 

Built a select input component that takes an array of options and a callback function. 

I wasn't sure about how to set up the functionality of this component. Should the callback function take this.state.value as a prop?

### CODE REVIEW

**Comment by torykling** [comment]
@ZakkMan I deleted the unused lifecycle methods, made options an array of objects instead of strings, and passed the target value as a prop to the onSelect method. 

Remaining questions:
I added proptypes and made onSelect a required prop because I kept getting error messages when I tried to set a method as a default onSelect prop. Is there a better way to handle this? 

Does the rest of the logic make sense?

As for styling, I tweaked it a bit but not much.  I know the font isn't right but I listed it as Reem Kufi and was hoping we'd import that font on the index.html page in the public folder, which I don't see here. Is there another way to link fonts?  Is there something else I should do?

File: client/src/components/InputSelect/InputSelect.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.InputSelect {
+  width: 20rem;
+  height: 3rem;
+  background: #fafafa;
+  border: 2px solid #e5e5e5;
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.css]: For colors, if you update your branch with `develop`, you should get the CSS variables that Bill created. Do you mind using those for the color and font properties?

File: client/src/components/InputSelect/InputSelect.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from "react";
+import "./InputSelect.css";
+import PropTypes from "prop-types";
+import "../../App.css";
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.js]: You shouldn't need to do this anymore - Lorenzo made a PR that set this for storybook globally

File: client/src/components/InputSelect/InputSelect.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+// Class Based React Component
+class InputSelect extends Component {
+  constructor(props) {
+    super(props);
+    console.log(props);
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.js]: Can you delete any `console.log` statements?

File: client/src/components/InputSelect/InputSelect.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      classList: "InputSelect",
+      value: "select"
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    e.preventDefault();
+    this.setState({ value: e.target.value });
+    this.props.onSelect(e.target.value);
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.js]: We probably want this to be synchronous, which you like this:

```js
this.setState({ value: e.target.value }, () => this.props.onSelect(e.target.value));
```

File: client/src/components/InputSelect/InputSelect.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  options: [
+    { value: "option1", title: "Option 1" },
+    { value: "option2", title: "Option 2" }
+  ]
+};
+InputSelect.propTypes = {
+  optionPrefix: PropTypes.string,
+  options: PropTypes.array,
+  onSelect: PropTypes.func.isRequired
+};
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.js]: 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 🔥 
This is so dope!

File: client/src/components/InputSelect/InputSelect.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import "./InputSelect.css";
+import PropTypes from "prop-types";
+
+// Class Based React Component
+class InputSelect extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      classList: "InputSelect",
```
Comment (ZakkMan) [inline client/src/components/InputSelect/InputSelect.js]: If you just have this one class, then you can put the string in your JSX for the `className` property

**Comment by ZakkMan** [comment]
@torykling Can you update your branch with `develop` and resolve any merge conflicts

**Pull request merged**